data/tekton-pipelines/okd/windows10-installer.yaml

```diff
@@ -0,0 +1,470 @@
+                        "storage": "7Gi"
+                      }
+                    }
+                  },
+                  "source": {
+                    "http": {
+                      "url": "$(params.winImageDownloadURL)"
+                    }
+                  }
+                }
+              }
+        - name: deleteDisks
+          value: "true"
+        - name: disks
+          value:
+            - '{"name": "installcdrom", "cdrom": {"bus": "sata"}, "bootOrder": 1}'
+            - '{"name": "rootdisk", "disk": {"bus": "virtio"}, "bootOrder": 2}'
+            - '{"name": "virtiocontainerdisk", "cdrom": {"bus": "sata"}}'
+            - '{"name": "sysprep", "cdrom": {"bus": "sata"}}'
+        - name: deleteVolumes
+          value: "true"
+        - name: volumes
+          value:
+            - '{"name": "installcdrom", "dataVolume": {"name": "${NAME}"}}'
+            - '{"name": "rootdisk", "dataVolume": {"name": "${BASEDV}"}}'
+            - '{"name": "virtiocontainerdisk", "containerDisk": {"image": "$(params.virtioContainerDiskName)"}}'
+            - '{"name": "sysprep", "sysprep": {"configMap": {"name": "$(params.autounattendConfigMapName)"}}}'
+        - name: deleteTemplateParameters
+          value: "true"
+        - name: templateParameters
+          value:
+            - '{"name": "NAME", "description": "VM name", "generate": "expression", "from": "windows-[a-z0-9]{6}"}'
+            - '{"name": "BASEDV", "description": "Name of DataVolume to install into"}'
+      runAfter:
+        - create-base-dv
+        - copy-template-installer
+      timeout: 10m
+      taskRef:
+        kind: ClusterTask
+        name: modify-vm-template
+    - name: create-vm-from-template
+      params:
+        - name: templateName
+          value: $(tasks.modify-vm-template-installer.results.name)
+        - name: templateParams
+          value:
+            - BASEDV:$(tasks.create-base-dv.results.name)
+        - name: runStrategy
+          value: RerunOnFailure
+        - name: startVM
+          value: "true"
+      runAfter:
+        - modify-vm-template-installer
+      taskRef:
+        kind: ClusterTask
+        name: create-vm-from-template
+    - name: wait-for-vmi-status
+      params:
+        - name: vmiName
+          value: $(tasks.create-vm-from-template.results.name)
+        - name: successCondition
+          value: status.phase == Succeeded
+        - name: failureCondition
+          value: status.phase in (Failed, Unknown)
+      runAfter:
+        - create-vm-from-template
+      timeout: 2h
+      taskRef:
+        kind: ClusterTask
+        name: wait-for-vmi-status
+    - name: cleanup-vm
+      params:
+        - name: vmName
+          value: $(tasks.create-vm-from-template.results.name)
+        - name: delete
+          value: "true"
+      runAfter:
+        - wait-for-vmi-status
+      timeout: 10m
+      taskRef:
+        kind: ClusterTask
+        name: cleanup-vm
+    - name: copy-template-golden
```

Comment:
This is a nice approach. I just wonder, if we could have the costomized templates as an result of PR " Add Windows 10 sysprep example pipeline https://github.com/kubevirt/tekton-tasks-operator/pull/28 " and re-adjust this pipeline to a "create windows boot source" example, if technically possible.

Reply:
I changed this PR so at the end of the pipeline a bootsource is created. :+1: